[macOS] Allow syscall used during launch of the WebContent process

pvollan opened the pull request:
#### cbded7d2834291b7f8d82f5783dadca70c71696d
<pre>
[macOS] Allow syscall used during launch of the WebContent process
<a href="https://bugs.webkit.org/show_bug.cgi?id=242218">https://bugs.webkit.org/show_bug.cgi?id=242218</a>
&lt;rdar://95210336&gt;

Reviewed by Chris Dumez and Alexey Proskuryakov.

Add SYS_quotactl to the WebContent process&apos; sandbox on macOS. It is only required on Intel during launch, and will be blocked post launch.

* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: <a href="https://commits.webkit.org/252054@main">https://commits.webkit.org/252054@main</a>
</pre>

aproskuryakov commented on Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
```diff
@@ -1873,6 +1873,7 @@
         SYS_getrlimit
         SYS_kdebug_trace
         SYS_pathconf
+        SYS_quotactl
```
Is this only needed for WebConent, and not any other services?

pvollan replied:
> Is this only needed for WebConent, and not any other services?

That is a good point! I will look into that. I have also updated the PR, since this sys call is only needed on Intel.

Thanks for reviewing!

cdumez reviewed (approved):
LGTM

pvollan committed:
[macOS] Allow syscall used during launch of the WebContent process
https://bugs.webkit.org/show_bug.cgi?id=242218
<rdar://95210336>

Reviewed by Chris Dumez and Alexey Proskuryakov.

Add SYS_quotactl to the WebContent process' sandbox on macOS. It is only required on Intel during launch, and will be blocked post launch.

* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: https://commits.webkit.org/252054@main
